File: Leetcode/string/520_Detect Capital/Solution.java
class Solution {
    public boolean detectCapitalUse(String word) {
        if(word==null || word.isEmpty()) return false;
        boolean res = Character.isUpperCase( word.charAt(0) ),
                res2 = !res, res3 = res;
        for(int i = 1; i<word.length(); i++){
            res3 &= Character.isLowerCase( word.charAt(i) );
            res2 &= Character.isLowerCase( word.charAt(i) );
            res &= Character.isUpperCase( word.charAt(i) );
        }
        return res || res2 || res3;
    }
}

File: Leetcode/string/520_Detect Capital/Solution.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        if (word.size() <= 1) return true;
        bool all_lower = true, all_upper = true;
        for(int i = 1; i < word.size(); ++i)
        {
                all_upper &= (isupper(word[i])!=0);
                all_lower &= (islower(word[i])!=0);
        }
        return all_lower || (all_upper && isupper(word[0]));
    }
};
